tools/cli/main.cpp

```diff
@@ -12,246 +12,859 @@
+                                                          SYM_TABLE_VERSION_GLOSSARY);
+
+static struct arg_file *xtract_input_fs = arg_filen(NULL, NULL, FILE_DATATYPE, 0, MAX_NO_INPUTS, INPUT_GLOSSARY);
+
+static struct arg_end *xtract_end = arg_end(20);
+
+static void *xtract_argtable[] =
+        {xtract, xtract_out_f, xtract_err_f, xtract_output_type, xtract_catalog_fs, xtract_symtable_name,
+         xtract_symtable_version, xtract_input_fs, xtract_end};
+
+static int xtract_nerrors;
+
+static void **all_arg_tables[] = {argtable, proc_argtable, comp_argtable, xtract_argtable, help_argtable,
+                                 version_argtable};
+static int all_arg_tables_size = sizeof(all_arg_tables) / sizeof(all_arg_tables[0]);
+
+//
+// Printing functions
+//
+
+void ion_print_process_examples() {
+    std::string s =
+            //--------------------------------------------------------------------------------
+            //<-                                80 chars                                    ->
+            //--------------------------------------------------------------------------------
+            "Read input.10n and pretty-print it to stdout.\n"
+            "\t $ ion process input.10n\n\n"
+
+            "Read input.ion (using a catalog comprised of the shared symbol tables contained \n"
+            "in catalog.10n) without re-writing, and write a performance report.\n"
+            "\t$ ion process --output-format none --catalog catalog.10n --perf-report report -- input.10n\n\n"
+
+            "Read input.10n according to the ReadInstructions specified by instructions.ion \n"
+            "and write the resulting Events to output.ion.\n"
+            "\t$ ion process -o output.ion -f events -t instructions.ion input.10n\n\n"
+
+            "Read input1.ion and input2.10n and output to stdout any values in the streams \n"
+            "that match the filter .foo.\n"
+            "\t$ ion process --filter .foo input1.ion input2.10n\n\n";
+
+    std::cout << s << std::endl;
+
 }
 
-static bool ion_cli_has_value(std::map<std::string, docopt::value> *args, const std::string arg) {
-    auto it = args->find(arg);
-    if (it == args->end()) return false;
-    docopt::value val = it->second;
-    return val.isString() || val.isBool() || val.isLong() || val.isStringList(); // Otherwise, the value is Empty.
+void ion_print_compare_examples() {
+    std::string s =
+            //--------------------------------------------------------------------------------
+            //<-                                80 chars                                    ->
+            //--------------------------------------------------------------------------------
+            "Compare each stream in read_events.ion, input1.ion, and input2.10n against all \n"
+            "other streams in the set and output a ComparisonReport to \n"
+            "comparison_report.ion.\n"
```

Comment:
looks like this might fit at the end of the previous line?

Reply:
Yeap.